test/unit/lib/plugins/aws/package/compile/layers.test.js

```diff
@@ -393,33 +128,96 @@ describe('lib/plugins/aws/package/compile/layers/index.test.js', () => {
         awsRequestStubMap,
       });
       expect(firstCfResources).to.have.property(firstLayerResourceName);
-      // Replaces
-      // https://github.com/serverless/serverless/blob/e78f695004bf292c4163daf9705e5e0c6cbe2592/test/unit/lib/plugins/aws/package/compile/layers/index.test.js#L147-L329
     });
   });
 
-  it.skip('TODO: should generate expected permissions resource', () => {
-    // Replaces
-    // https://github.com/serverless/serverless/blob/e78f695004bf292c4163daf9705e5e0c6cbe2592/test/unit/lib/plugins/aws/package/compile/layers/index.test.js#L331-L383
+  it('should generate expected permissions resource', () => {
+    const layerResourceName = naming.getLambdaLayerLogicalId('LayerOne');
+    const layerOne = cfResources[layerResourceName];
+    const s3Folder = service.package.artifactDirectoryName;
+    const s3FileName = service.layers.layerOne.package.artifact.split(path.sep).pop();
+
+    expect(layerOne.Type).to.be.equals('AWS::Lambda::LayerVersion');
+    expect(layerOne.Properties.Content.S3Key).to.be.equals(`${s3Folder}/${s3FileName}`);
+    expect(layerOne.Properties.Content.S3Bucket.Ref).to.be.equals('ServerlessDeploymentBucket');
+    expect(layerOne.Properties.LayerName).to.be.equals('layerOne');
+
+    const description = 'Current Lambda layer version';
+    expect(cfOutputs.LayerLambdaLayerQualifiedArn.Description).to.be.equals(description);
+    expect(cfOutputs.LayerLambdaLayerQualifiedArn.Value.Ref).to.be.equals('LayerLambdaLayer');
+
+    const layerNamePermission = naming.getLambdaLayerPermissionLogicalId('LayerOne', 'Wild');
+    const layerPermission = cfResources[layerNamePermission];
+
+    expect(layerPermission.Type).to.be.equals('AWS::Lambda::LayerVersionPermission');
+    expect(layerPermission.Properties.Action).to.be.equals('lambda:GetLayerVersion');
+    expect(layerPermission.Properties.LayerVersionArn.Ref).to.be.equals('LayerOneLambdaLayer');
+    expect(layerPermission.Properties.Principal).to.be.equals('*');
   });
 
-  it.skip('TODO: should support `layers[].allowedAccounts`', () => {
-    // Replaces
-    // https://github.com/serverless/serverless/blob/e78f695004bf292c4163daf9705e5e0c6cbe2592/test/unit/lib/plugins/aws/package/compile/layers/index.test.js#L331-L452
+  it('should support `layers[].allowedAccounts`', () => {
+    const layerResourceName = naming.getLambdaLayerLogicalId('LayerTwo');
```

Comment:
Here also let's just validate support for `allowedAccounts` and remove all assertions that were already performed in previous tests